src/lib/utils/round-weights.ts

```diff
@@ -1,4 +1,11 @@
-import { constants, type SplitsReceiverStruct } from 'radicle-drips';
+import { BigNumber } from 'ethers';
+import { constants } from 'radicle-drips';
+
+export interface TypedReceiver {
+  typeOfReceiver: 'Dependency' | 'Maintainer';
+  accountId: string;
+  weight: number;
+}
```

Comment:
again a bit of a nitpick, but if you type `roundWeights` as `roundWeights<T extends { accountId: string, weight: number }>(receivers: T[]): T[]`, you could pass any object in that has at least accountId and weight, and the return type would still include any additional fields (e.g. `typeOfReceiver` here).

would just make it a bit more generic 🤷